geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ClientUserAuths.java

```diff
@@ -99,19 +98,20 @@ public Subject getSubject(final Long userId) {
     return uniqueIdVsSubject.get(userId);
   }
 
-  public boolean removeSubject(final Long userId) {
-    final Subject subject = uniqueIdVsSubject.remove(userId);
+  public void removeSubject(final Long userId) {
     logger.debug("Subject of {} removed.", userId);
+    removeSubject(uniqueIdVsSubject.remove(userId));
+  }
+
+  @VisibleForTesting
+  void removeSubject(Subject subject) {
```

Comment:
I would really like to see the increased use of `@Nullable`, and `@NotNull`, annotation where appropriate.